graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -268,47 +303,128 @@ def enter(self, node, key, parent, path, ancestors):
             renamed_node = self._rename_name_and_add_to_record(node)
             if renamed_node is node:  # Name unchanged, continue traversal
                 return None
-            else:  # Name changed, return new node, `visit` will make shallow copies along path
+            else:
+                # Name changed or suppressed, return new node, `visit` will make shallow copies
+                # along path
                 return renamed_node
         else:
             # All Node types should've been taken care of, this line should never be reached
             raise AssertionError('Unreachable code reached. Missed type: "{}"'.format(node_type))
 
 
 class RenameQueryTypeFieldsVisitor(Visitor):
-    def __init__(self, renamings, query_type):
+    def __init__(self, renamings: Dict[str, Optional[str]], query_type: str):
         """Create a visitor for renaming fields of the query type in a schema AST.
 
         Args:
-            renamings: Dict[str, str], from original field name to renamed field name. Any
+            renamings: Dict[str, Optional[str]], from original field name to renamed field name. Any
                        name not in the dict will be unchanged
             query_type: str, name of the query type (e.g. RootSchemaQuery)
         """
         # Note that as field names and type names have been confirmed to match up, any renamed
-        # field already has a corresponding renamed type. If no errors, due to either invalid
-        # names or name conflicts, were raised when renaming type, no errors will occur when
+        # query type field already has a corresponding renamed type. If no errors, due to either
+        # invalid names or name conflicts, were raised when renaming type, no errors will occur when
         # renaming query type fields.
         self.in_query_type = False
         self.renamings = renamings
         self.query_type = query_type
 
-    def enter_object_type_definition(self, node, *args):
+    def enter_object_type_definition(self, node: ObjectTypeDefinitionNode, *args):
         """If the node's name matches the query type, record that we entered the query type."""
         if node.name.value == self.query_type:
             self.in_query_type = True
 
-    def leave_object_type_definition(self, node, key, parent, path, ancestors):
+    def leave_object_type_definition(
+        self, node: ObjectTypeDefinitionNode, key, parent, path, ancestors
+    ):
         """If the node's name matches the query type, record that we left the query type."""
+        if not node.fields:
+            raise CascadingSuppressionError(
+                f"Type renamings {self.renamings} suppressed every type in the schema so it will be"
+                f" impossible to query for anything. To fix this, check why the `renamings` "
+                f"argument of `rename_schema` mapped every type to None. Note that adding "
```

Comment:
The other two existing suppression error messages have an easily actionable step: add more suppressions so that the field/union doesn't end up in an illegal state. It seems really unlikely that someone would want to suppress every type in practice, because then you'd have a useless (empty) schema, so I'm not sure what makes the most sense to write here.

Reply:
Yeah this is fair, I wouldn't worry about it too much. The thing I wanted to ask about is this: why is this a `Cascading SuppressionError`? I don't see the cascading aspect of it, so why not use the more general `SchemaTransformError`?

Reply:
Right now we only use `SchemaTransformError` as a parent class for more specific errors like `CascadingSuppressionError`-- the code raises only the specific errors and not `SchemaTransformError`. Given that there's only one way to suppress every type, does it make more sense to make a whole new error class just for this or just use the existing `SchemaTransformError` parent class? I suppose it's reasonably clear from the error message what the problem is, but I suppose it might depend on what's more convenient to see at a glance?